WorldRestoration/src/main/java/as/minecraft/worldrestoration/dependencies/CoreProtectHook.java

```diff
@@ -0,0 +1,44 @@
+package as.minecraft.worldrestoration.dependencies;
+
+import org.bukkit.Bukkit;
+import org.bukkit.plugin.Plugin;
+
+import net.coreprotect.CoreProtect;
+import net.coreprotect.CoreProtectAPI;
+
+public class CoreProtectHook {
+	public static CoreProtectAPI coreProtect;
+	
+	public CoreProtectHook(){
+		CoreProtectHook.coreProtect = this.fetchCoreProtect();
+	}
+	
+	public CoreProtectAPI getCoreProtect() {
+		return CoreProtectHook.coreProtect;
+	}
+	
+	private CoreProtectAPI fetchCoreProtect() {
+	    Plugin plugin = Bukkit.getServer().getPluginManager().getPlugin("CoreProtect");
+	 
+	    // Check that CoreProtect is loaded
+	    if (plugin == null || !(plugin instanceof CoreProtect)) {
+	    	//Bukkit.broadcastMessage("CoreProtect not loaded!");
+	        return null;
+	    }
+
+	    // Check that the API is enabled
+	    CoreProtectAPI CoreProtect = ((CoreProtect) plugin).getAPI();
+	    if (CoreProtect.isEnabled() == false) {
+	    	//Bukkit.broadcastMessage("API not enabled");
+	        return null;

```

Comment:
Jeg liker ikke `null` (og det gjør ikke programmerings-miljøet heller). Hvem har vel ikke sett `NullPointerException` før? Tidligere kollega snakker om det her: https://www.youtube.com/watch?v=FTfLLB4kboQ

Ofte ønsker man å representere noe med `null`. I listen er det noen eksepler, medfølgende et alternativ på hva som ofte blir betraktet som en bedre metode:
- Variablen kan mangle verdi, ikke være initialisert etc. : Her er vel Optional bedre.
- Funksjon kan la være å returnere: Optional er bedre
- Man kan slippe å sette argumenter: Optional er nok bedre
- Noe gikk galt, og jeg fikk ikke tak i verdien: Exception eller Optional er vel foretrukket her.
- etc.

Er en vanskelig diskusjon dette. Hvis det er forventet at man ikke har en verdi ville jeg nok gått for optional, og spart exception til mer uventede feil. Hvis `CoreProtect` alltid er forventet å starte først er exception best. Hvis f.eks rekkefølgen er mer tilfeldig, og du vil signalisere at man bare skal vente litt kan man bruke Optional, eller en egen wrapper klasse som sier i kode hva problemet er, og hva man skal gjøre med det.

Reply:
Men Optional og Exception står det i "kontrakten" til funksjonen at den ikke lover å returnere et svar, eller at feil kan forventes. Hvis man returnerer `null` er det ingen ting i funksjonsdefinisjonen man kan se som tilsier at noe kan ha gått galt. Hvis jeg leser en funksjon som bare returnerer en verdi antar jeg egentlig at den ikke er `null`, mens med optional for eksempel vil koden kreve at jeg håndterer tilfellet med at det er null, og det er klart signalisert i funksjonsdefinisjonen.

Reply:
De kan også kombineres selvfølgelig. Kan hende enkelte tilfeller tilsier at det skal være optional (f.eks. at plugin er Ok, men ikke lastet inn enda, eller at det ikke er den pluginen etc.) og exception til mer alvorlige og uventede feil som "Wrong API version".

Reply:
Will do.

Reply:
Mener du her å bruke throwe en exception? Type IllegalStateException i stedet for return null? Eller er det noen en annen retur-type?

```suggestion
private CoreProtectAPI fetchCoreProtect() {
	    Plugin plugin = Bukkit.getServer().getPluginManager().getPlugin("CoreProtect");
	    	// Check that CoreProtect is loaded
	    	if (plugin == null || !(plugin instanceof CoreProtect)) {
	    		Bukkit.getLogger().severe("Could not load CoreProtect, make sure that you have the latest version of CoreProtect installed!");
	    		this.plugin.getPluginLoader().disablePlugin(this.plugin);
	    	}

	    	// Check that the API is enabled
	    	CoreProtectAPI CoreProtect = ((CoreProtect) plugin).getAPI();
	    	if (CoreProtect.isEnabled() == false) {
	    		throw new IllegalStateException();
	    	}

	    	// Check that a compatible version of the API is loaded
	    	if (CoreProtect.APIVersion() < 6) {
	    		Bukkit.getLogger().severe("Could not load CoreProtect, make sure that you have the latest version ");
	    		throw new IllegalStateException();
	    	}
	    	
	    	return CoreProtect;
	}
```